lib/galaxy/datatypes/sequence.py

```diff
@@ -1066,3 +1066,22 @@ def sniff(self, filename):
 
         # Number of lines is less than 3
         return False
+
+class Genbank(data.Text):
+    """Class representing a Genbank sequence"""
+    edam_format = "format_1936"
+    file_ext = "genbank"
+
+    def sniff(self, filename):
+        try:
+            fh = open(filename, 'r')
+            while True:
+                line = fh.readline()
+                if not line:
+                    break  # EOF
+                line = line.strip()
+                return line.startswith('LOCUS ')
```

Comment:
You are effectively reading only the first line, you can replace the whole `while` cycle with:
```python
            line = fh.readline().strip()
            return line.startswith('LOCUS ')
```